.github/workflows/build.yml

```diff
@@ -0,0 +1,81 @@
+name: Build
+
+on: [push]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          # - macOS-latest
+
+    steps:
+    # Install Dependencies
+    - name: Ubuntu Dependencies
+      if: startsWith(matrix.os,'ubuntu')
+      run: |
+          sudo apt-get install -y ninja-build
+          sudo apt-get install -y llvm-dev
+          sudo apt-get install -y libclang-dev
+
+    - name: Windows Dependencies
+      if: startsWith(matrix.os,'windows')
+      run: |
+          choco install -y ninja
+          choco install -y llvm
+
+    - name: MacOS Dependencies
+      if: startsWith(matrix.os,'macos')
+      run: |
+          brew install ninja
+
+    - name: Extract the version of Clang
+      run: |
+        echo ::set-env name=VERSION::$( clang --version | sed -n 's/^.*version *\([0-9]*.[0-9]\).*$/\1/p' )
+        echo ::set-env name=VERSION_int::$( clang --version | sed -n 's/^.*version *\([0-9]*\)..*$/\1/p' )
+
+    - name: Absolute paths to LLVM's root and bin directory
+      run: |
+        echo ::set-env name=PREFIX_PATH::$( llvm-config-$VERSION_int --prefix )
+        echo ::set-env name=BIN_PATH::$( llvm-config-$VERSION_int --bindir )
+
+    # Checkout proper Branch
```

Comment:
This seems backwards -- it will conditionally build different IWYU versions depending on which LLVM/Clang packages were available.

So if I push a change to current `master` (which corresponds to Clang 11), and Clang 11 packages are not available for whatever reason, it will check out the version it managed to install. That seems like surprising behavior.

Reply:
> Judging from the error message, I would guess that you're missing one or more clang packages -- the packaging for Ubuntu is a little wonky, so the Clang cmake configs come with one package, and the binaries with another, if I remember correctly. Check the travis config for the definitive package deps: [/.travis.yml@`master`](https://github.com/include-what-you-use/include-what-you-use/blob/master/.travis.yml) [](/include-what-you-use/include-what-you-use/blob/HEAD@{2020-05-22T09:24:06Z}/.travis.yml)

That is why I think providing binaries is essential.  Some of the packages like `llvm-dev` don't exist on Windows altogether. Also, building LLVM from the source requires 40GB of space and takes many hours.

If we cannot set up the environment in this script that means not many people can use `include-waht-you-use`.

What is the command-line form of this part?
```
- sourceline: 'deb http://apt.llvm.org/bionic llvm-toolchain-bionic main'
       key_url: 'apt.llvm.org/llvm-snapshot.gpg.key'
```

Reply:
That sets up the apt repository and trusts its public key, so I would guess it's something like:
```
$ sudo apt-add-repository ...
$ sudo apt-add-key ...
``` 
I think there are some scattered instructions on https://apt.llvm.org/.

Reply:
Let me try their automatic bash script to see if that helps.

Reply:
OK finally I got this working in https://github.com/include-what-you-use/include-what-you-use/pull/799/commits/8ce344d8f4b22018bc75f153df3e16add7ee278c

We might have to write some conditions for LLVMs older than 8 or for LLVM11.